### PR TITLE
Fix renewal error in the `product-move` API

### PR DESCRIPTION
**Pull request by jonflynng**
Term renewal for many subs happens during the billing run on the renewal day which is scheduled for around 6am BST.

During this billing run, Zuora does not return the contribution invoice item, only supporter plus invoice items. This PR ensures this edge case is still processed.

### CODE REVIEW

File: handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/BuildPreviewResult.scala [hunk 1]
```diff
@@ -34,10 +38,36 @@ object BuildPreviewResult {
           supporterPlusInvoiceItems.head.totalAmount,
           supporterPlusInvoiceItems(1).serviceStartDate,
         )
+      /*
+              Term renewal for many subs happens during the billing run on the renewal day which is scheduled for around 6am BST.
+              During this billing run, Zuora does not return the contribution invoice item, only supporter plus invoice items.
+              This ensures this edge case is still processed.
+       */
+      case (n1, n2) if n1 > 1 && n2 == 0 =>
```
Comment (jonflynng) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/BuildPreviewResult.scala:46]: @rupertbates  I'm going to add a test for this and think about any edge cases. This is the logic.

**Comment by jonflynng** [comment]
[Fix product-move API error when customer switches during billing run on renewal date](https://trello.com/c/Iddq3GjX/4139-fix-product-move-api-error-when-customer-switches-during-billing-run-on-renewal-date)

File: handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/BuildPreviewResult.scala [hunk 2]
```diff
@@ -34,10 +38,36 @@ object BuildPreviewResult {
           supporterPlusInvoiceItems.head.totalAmount,
           supporterPlusInvoiceItems(1).serviceStartDate,
         )
+      /*
+              Term renewal for many subs happens during the billing run on the renewal day which is scheduled for around 6am BST.
+              During this billing run, Zuora does not return the contribution invoice item, only supporter plus invoice items.
+              This ensures this edge case is still processed.
```
Comment (rupertbates) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/BuildPreviewResult.scala:44]:
```suggestion
              When this happens we can just use the full price of the active rate plan as the amount to be refunded
```

File: handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/BuildPreviewResult.scala [hunk 3]
```diff
@@ -34,10 +38,36 @@ object BuildPreviewResult {
...
+       */
+      case (n1, n2) if n1 > 1 && n2 == 0 =>
+        for {
+          date <- Clock.currentDateTime.map(_.toLocalDate)
+          isRenewalDate = activeRatePlanCharge.effectiveStartDate == date
+          priceDifference = if (isRenewalDate) activeRatePlanCharge.price.get else 0
```
Comment (rupertbates) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/BuildPreviewResult.scala:50]:
```suggestion
          priceDifference = if (isRenewalDate) activeRatePlanCharge.price.getOrElse(0) else 0
```
A little bit safer

**Review by rupertbates — APPROVED** [review]
Couple of small suggestions, otherwise good to go
